src/Diagnostics.RuntimeHost/Startup.cs

```diff
@@ -86,6 +86,9 @@ public void ConfigureServices(IServiceCollection services)
             ChangeAnalysisTokenService.Instance.Initialize(dataSourcesConfigService.Config.ChangeAnalysisDataProviderConfiguration);
             AscTokenService.Instance.Initialize(dataSourcesConfigService.Config.AscDataProviderConfiguration);
             CompilerHostTokenService.Instance.Initialize(Configuration);
+
+            // Initialize on startup
+            servicesProvider.GetService<ISourceWatcherService>();
```

Comment:
How does this line start the source watcher to download the detectors?

Reply:
This line forces the initializes of the service Singleton SourceWatcherService which upon initialization will in turn initialize a GitHubWatcher or a LocalFileSystemWatcher.

The line more specifically requests the ISourceWatcherService and since the singleton has not been initialized yet it will initialize the ISourceWatcherService so it can return the ISourceWatcherService